google/api_versions.go

```diff
@@ -25,3 +31,63 @@ func Convert(item, out interface{}) error {
 
 	return nil
 }
+
+type TerraformResourceData interface {
+	HasChange(string) bool
+	GetOk(string) (interface{}, bool)
+}
+
+// Compare the keys set in schema against a list of features and their versions to determine
+//what version of the API is required in order to manage the resource.
+func getComputeApiVersion(d TerraformResourceData, baseVersion ComputeApiVersion, keys []Key) ComputeApiVersion {
```

Comment:
What's the purpose of passing in a base version? Is it just for testing or would it ever not be `v1` in practice?